src/Mod/TechDraw/Gui/Command.cpp

```diff
@@ -357,64 +408,99 @@ void CmdTechDrawView::activated(int iMsg)
         }
     }
 
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/TechDraw");
     if (shapes.empty() && xShapes.empty()) {
-        QMessageBox::warning(Gui::getMainWindow(), QObject::tr("Wrong selection"),
-                             QObject::tr("No Shapes, Groups or Links in this selection"));
-        return;
-    }
+        if (!viewCreated) {
+            // If nothing was selected, then we offer to insert SVG or Images files.
+            bool dontShowAgain = hGrp->GetBool("DontShowInsertFileMessage", false);
+            if (!dontShowAgain) {
+                QMessageBox msgBox;
+                msgBox.setText(msgBox.tr("Note: If you want to insert a shape you need to select it before starting the tool, else it opens a file browser to insert SVG or Images."));
```

Comment:
Maybe: 
"If you want to insert a view from existing objects, please select them before evoking this tool. Without a selection, a file browser will open, to insert a SVG or image file."

As it's not only shapes but also spreadsheets, arch, 2D faces,...
also just image, multi selection is not possible in the file browser on my side.

Is it possible to have a "go back" button as well, if a user mistakenly did not have a selection and just wants to abort and select something? Currently there is only OK and then the file browser opens.